flipperdevices/flipperzero-firmware
# NFC: Support reading Mifare Classic key B from sector trailer, reading sector with B key where A key can't read block,  Nfc Magic app not using NFC folder by default (in file select)

# What's new

- NFC: Support reading Mifare Classic key B from sector trailer and reusing it for other sectors

- NFC: Fix reading sector with B key where A key can't read block (fixes #2413)

- NFC: Fix Nfc Magic app not using NFC folder by default (in file select)

Notes: 
I modified `mf_classic_auth_attempt` because it isn't used anywhere and it was easy to do what I needed (and didn't break anything, but actually improved it)

# Verification 

1. 
- Write sector with access bits that allow reading key B, example:
Block 3: `A0A1A2A3A4A5FF0780001122FF112299`
Block 7: `1122FF112299778F0800FF11223344AA`
- Try reading tag with Flipper, it should read 3 keys and 2 sectors

2.
- From #2423 (that is duplicate of #2413)
> - Flash [ac.zip](https://github.com/flipperdevices/flipperzero-firmware/files/10820730/ac.zip) to an open or magic tag.
> - Use `NFC -> Extra Actions -> Read Specific Card Type -> Read Mifare Classic` to read the tag.
- Tag should be read fully

3.
- Run Nfc Magic app, select Write Gen1a. File selector should be in NFC folder
# Checklist (For Reviewer)
